crypto/signals.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2019-2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+
+#include <openssl/crypto.h>
+#include <internal/list.h>
+
+#define CRYPTO_SIGNAL_IS_DEFAULT_ACTION(X) (X == NULL)
+
+static struct list siglist;
```

Comment:
Hmm. Global variable. I see no locking so by implication all functions accessing this are not thread safe. This needs to reflected in the public API docs.

Reply:
How is the siglist freed on close down?

Reply:
Added a call to `OPENSSL_atexit` (l.92).